include/faabric/util/snapshot.h

```diff
@@ -67,6 +62,19 @@ class SnapshotDiff
     }
 };
 
+class SnapshotMergeRegion
+{
+  public:
+    uint32_t offset = 0;
+    size_t length = 0;
+    SnapshotDataType dataType = SnapshotDataType::Raw;
+    SnapshotMergeOperation operation = SnapshotMergeOperation::Overwrite;
+
+    void addDiffs(std::vector<SnapshotDiff>& diffs,
+                  const uint8_t* original,
+                  const uint8_t* updated);
+};
```

Comment:
Moved this class down the file as it now depends on the `SnapshotDiff` class which is declared above.